components/async-status-mgt/org.wso2.carbon.identity.framework.async.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/status/mgt/dao/AsyncStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,880 @@
+import static org.wso2.carbon.identity.framework.async.status.mgt.util.Utils.isMSSqlDB;
+
+/**
+ * DAO implementation for Asynchronous Operation Status Management.
+ */
+public class AsyncStatusMgtDAOImpl implements AsyncStatusMgtDAO {
+
+    //TODO: apache commons log
+    private static final Logger LOGGER =
+            Logger.getLogger(AsyncStatusMgtDAOImpl.class.getName());
+
+    @Override
+    public String registerAsyncOperationWithoutUpdate(OperationRecord record) {
+
+        String generatedOperationId;
+        String currentTimestamp = new Timestamp(new Date().getTime()).toString();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                     CREATE_ASYNC_OPERATION_IDN, IDN_OPERATION_ID)) {
+
+            statement.setString(SQLConstants.OperationStatusTableColumns.IDN_CORRELATION_ID, record.getCorrelationId());
+            statement.setString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_TYPE, record.getOperationType());
+            statement.setString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_SUBJECT_TYPE,
+                    record.getOperationSubjectType());
+            statement.setString(
+                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_SUBJECT_ID, record.getOperationSubjectId());
+            statement.setString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_INITIATED_ORG_ID,
+                    record.getResidentOrgId());
+            statement.setString(
+                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_INITIATED_USER_ID, record.getInitiatorId());
+            statement.setString(
+                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_STATUS, OperationStatus.ONGOING.toString());
+            statement.setString(SQLConstants.OperationStatusTableColumns.IDN_CREATED_AT, currentTimestamp);
+            statement.setString(SQLConstants.OperationStatusTableColumns.IDN_LAST_MODIFIED, currentTimestamp);
+            statement.setString(
+                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_POLICY, record.getOperationPolicy());
+
+            int rowsAffected = statement.executeUpdate();
+            LOGGER.info("Async Operation Registering Success. Rows affected: " + rowsAffected);
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                generatedOperationId = generatedKeys.getString(1);
+                LOGGER.info("Generated IDN_OPERATION_ID: " + generatedOperationId);
+            } else {
+                throw new SQLException("Creating operation failed, no ID obtained.");
+            }
+        } catch (SQLException e) {
+            String errorMessage =
+                    "Error while registering the asynchronous operation";
+            throw new RuntimeException(errorMessage, e);
```

Comment:
Rather than throwing the base RuntimeException, let's create a specific runtime exception class specific for this service and use it here.